tests/selenium/conftest.py

```diff
@@ -0,0 +1,196 @@
+import pytest
+
+from dash.testing.browser import Browser
+from preconditions import preconditions
+from selenium.webdriver.common.action_chains import ActionChains
+from selenium.webdriver.common.by import By
+from selenium.webdriver.common.keys import Keys
+from selenium.webdriver.support import expected_conditions as EC
+from selenium.webdriver.support.wait import WebDriverWait
+
+_validate_col = lambda col: (isinstance(col, str) and len(col) > 0) or (isinstance(col, int) and col >= 0)
+_validate_col_id = lambda col_id: isinstance(col_id, str) and len(col_id) > 0
+_validate_id = lambda id: isinstance(id, str) and len(id) > 0
+_validate_key = lambda key: isinstance(key, str) and len(key) == 1
+_validate_row = lambda row: isinstance(row, int) and row >= 0
+_validate_state = lambda state: state in [_READY, _LOADING, _ANY]
+_validate_target = lambda target: isinstance(target, DataTableFacade)
+_validate_mixin = lambda mixin: isinstance(mixin, DataTableMixin)
```

Comment:
Use preconditions to validate the inputs. The param name in the lambda needs to match the param name in the function, hence all keys should be `key`, etc.